_books/ion-1-1/src/macros/macros_by_example.md

```diff
@@ -155,720 +132,160 @@ The outer e-expression receives the results from the expansion of the inner e-ex
-specific type or shape. We call Ion “self-describing” precisely because each value
-self-describes its type via a type tag.
-
-If schema constraints are enforced through some mechanism outside the serializer/deserializer,
-the type tags are unnecessary and may add up to a non-trivial amount of wasted space.
-Furthermore, the overhead for each value also includes length information: encoding an
-octet of data takes two octets on the stream.
-
-Ion 1.1 tries to mitigate this overhead in the binary format by allowing macro parameters to use
-more-constrained _tagless types_. These are subtypes of the concrete types,
-constrained such that type tags are not necessary in the binary form. In general this can shave
-4-6 bits off each value, which can add up in aggregate. In the extreme, that octet of data can
-be encoded with no overhead at all.
+## Available Tagless Types
 
 The following tagless types are available:
 
 | Tagless type                         | Description                          |
 |--------------------------------------|--------------------------------------|
-| `flex_symbol`                        | Tagless symbol (SID or text)         |
-| `flex_string`                        | Tagless string                       |
-| `flex_int`                           | Tagless, variable-width signed int   |
-| `flex_uint`                          | Tagless, variable-width unsigned int |
+| `int`                                | Variable-width signed integer        |
 | `int8`  `int16`   `int32`   `int64`  | Fixed-width signed int               |
+| `uint`                               | Variable-width unsigned integer      |
 | `uint8` `uint16`  `uint32`  `uint64` | Fixed-width unsigned int             |
 | `float16` `float32` `float64`        | Fixed-width float                    |
+| `symbol`                             | Symbol (SID or text)                 |
+| `string`                             | String                               |
+| `decimal`                            | Decimal number                       |
+| `timestamp`                          | Timestamp                            |
+| `blob`                               | Binary data                          |
+| `clob`                               | Character data                       |
 
+## Annotated Placeholders
 
-To define a tagless parameter, just declare one of the primitive types:
+Placeholders themselves can be annotated, which causes the expanded value to have that annotation:
 
 ```ion
-(macro point (flex_int::x flex_int::y)
-  {x: (%x), y: (%y)})
-```
-```ion
-(:point 3 17) ⇒ {x:3, y:17}
+(annotated_value [foo::(:?)])
 ```
 
-The tagless encoding has no real benefit here in text, as primitive types aim to improve the binary
-encoding.
-
-This density comes at the cost of flexibility. Primitive types cannot be annotated or null, and
-arguments cannot be expressed using macros, like we’ve done before:
```

Comment:
These sentences are still good to keep. (Although not in the middle of the annotated placeholders section.)